randovania/layout/game_patches_serializer.py

```diff
@@ -139,41 +149,33 @@ def _decode_pickup_assignment(
     locations: dict,
 ) -> PickupAssignment:
     """Decodes the `locations` key."""
-    target_name_re = re.compile(r"(.*) for Player (\d+)")
-
     pickup_assignment: PickupAssignment = {}
 
     initial_pickup_assignment = all_pools[player_index].assignment
 
-    for world_name, world_data in locations.items():
-        for area_node_name, target_name in typing.cast("dict[str, str]", world_data).items():
-            if target_name == _NOTHING_PICKUP_NAME:
-                continue
+    for location in locations:
+        pickup_name = location["pickup"]
+        if pickup_name == _NOTHING_PICKUP_NAME:
+            continue
 
-            pickup_name_match = target_name_re.match(target_name)
-            if pickup_name_match is not None:
-                pickup_name = pickup_name_match.group(1)
-                target_player = int(pickup_name_match.group(2)) - 1
-            else:
-                pickup_name = target_name
-                target_player = 0
+        target_player = location["owner"]
 
-            node_identifier = NodeIdentifier.create(world_name, *area_node_name.split("/", 1))
-            node = region_list.typed_node_by_identifier(node_identifier, PickupNode)
-            pickup: PickupEntry | None
+        node_identifier = NodeIdentifier.from_json(location["node_identifier"])
+        node = region_list.typed_node_by_identifier(node_identifier, PickupNode)
```

Comment:
should I assert here that node == location["index"] ?

Reply:
I'd actually just grab the node from the pickup index.

Reply:
question then still remains tho, should we ensure that when loading the layout, that the specified index and identifier are the same? Or should stuff like
```json
{
  "node_identifier": {
    "region": "Great Temple",
    "area": "Electric",
    "node": "Boogaloo"
  },
  "index": 1
}
```
pass?

Reply:
I would rather it just doesn't care, but maybe it's bad for people messing around? But I also would prefer no one to ever edit this file manually...

Reply:
yeah afaik its not toooo uncommon that people mess with it manually, and i'd rather have a sensible failsafe there then just in case someone does mess around with it.